### PR TITLE
p:d:T:copy_triangulation() for refined meshes. Second attempt.

### PR DESCRIPTION
**Pull request by marcfehling**
Alternative to #11947.

Instead of creating a coarse triangulation and reconstructing the refinement history, we can simply copy all data structures in `p:d:Triangulation`. This should be faster, especially in `MGTransferGlobalCoarsening`.

### CODE REVIEW

**Review by peterrum — COMMENTED** [review]
Looks good. Thanks! Could you add a test and run locally `ctest -R multigrid-global-coarsening`

**Comment by marcfehling** [comment]
The `multigrid-global-coarsening` tests have failed prior to your suggestion. Now they pass!

File: tests/mpi/tria_copy_triangulation_02.cc [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+template <int dim>
+void
+test()
+{
+  unsigned int myid = Utilities::MPI::this_mpi_process(MPI_COMM_WORLD);
+
+  parallel::distributed::Triangulation<dim> tria1(MPI_COMM_WORLD);
+  {
+    GridGenerator::hyper_cube(tria1);
```
Comment (peterrum) [inline tests/mpi/tria_copy_triangulation_02.cc:36]: Could you use here a geometry with a manifold attached to? Like hyper_ball? And maybe also set the boundary ids?

File: doc/news/changes/minor/20210319Fehling [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Improved: Function parallel::distributed::Triangulation::copy_triangulation()
```
Comment (peterrum) [inline doc/news/changes/minor/20210319Fehling:1]:
```suggestion
Improved: The function parallel::distributed::Triangulation::copy_triangulation()
```

**Review by peterrum — APPROVED** [review]

File: tests/mpi/tria_copy_triangulation_02.cc (hunk 1 above)
Comment (marcfehling) [reply tests/mpi/tria_copy_triangulation_02.cc]: I used `hyper_shell` with the `colorize` option and added a few assertions. Looks good to me now!

**Review by peterrum — APPROVED** [review]
Thank you!

**Review by kronbichler — APPROVED** [review]

**Pull request merged**